.github/workflows/formatting-and-linting.yml

```
@@ -4,23 +4,24 @@ on:
     branches-ignore:
       - main
 permissions:
-  contents: write
+  contents: read
```

```suggestion
  contents: write
```
This will need write permissions as it will commit to the branch as part of it's work

---

The job step has write permission, it doesn't need write at the workflow level. See below commits. (Note 1 failed because I pushed while it was running so it's push was rejected, not because of permissions)

---

You are correct